server/models/testResult.spec.js

```diff
@@ -0,0 +1,89 @@
+import test from 'ava';
+import request from 'supertest';
+import app from '../server';
+import TestResult from './testResult';
+import { connectDB, dropDB } from '../util/test-helpers';
+import bodyParser from 'body-parser';
+app.use(bodyParser.json());
+
+// Initial results added into test db
+const results = [
+  new TestResult({ testName: 'Passing Test', result: 'success', duration: 456, isStable: true }),
+  new TestResult({ testName: 'Failing Test', result: 'failure', message: 'something broke', duration: 123, isStable: false }),
+];
+
+test.before('connect to mockgoose', async () => {
+  await connectDB();
+});
+
+test.beforeEach('connect and add two result entries', async () => {
+  await TestResult.create(results).catch(() => 'Unable to create results');
+});
+
+test.afterEach.always(async () => {
+  await dropDB();
+});
+
+// From here
+test.serial('Should correctly give number of test results', async t => {
+  t.plan(2);
+
+  const res = await request(app)
+    .get('/api/results')
+    .set('Accept', 'application/json');
+
+  t.is(res.status, 200);
+  t.deepEqual(results.length, res.body.results.length);
+});
+
+test.serial('Should send correct data when queried against an ID', async t => {
+  t.plan(6);
```

Comment:
Full disclosure, I don't know what `.plan` does

Reply:
I didn't either and had to look it up.  `.plan()` is neat, it is the number of assertions to expect.  This is because of the async nature of these tests.  If you mess up and don't wait for the async call that contains the assertions, the test would normally complete before that code is reached and "pass", meanwhile the assertions that would have failed just never ran.  With `.plan()`, in that event you will get a failure saying something like "x# of assertions were expected, but only got 0"

Reply:
yoooooo. I'm down with that.